src/core/matter/mod.rs

```diff
@@ -5,14 +5,56 @@ use crate::error::{err, Error, Result};
 
 pub mod tables;
 
-pub trait Matter: Default {
+pub(crate) trait Matter: Default {
```

Comment:
If "Matter" truly exists in the CESR domain, and nowhere else. Let's start adding some comment headers for all "trait" classes 

Because we are using an english grammatical structure, including the etymological roots of the words would be ideal. 

I suggest we answer these questions for all "trait" classes. 
1. "What is a <trait>?"
2. "What are the etymological roots of <trait>?"

Reply:
Right, Artem made a suggestion that I misunderstood as backwards. I'm not sure this crate restriction makes sense, and have been thinking about multiple traits to break up the concerns and permit more flexibility in `parside`. Hiding some functionality with `pub(crate)` while exposing another set.

I think what you suggest is a great idea, but right now I am working for functionality to unblock other efforts in my company. If you want, the documentation is mostly there in KERIpy, and you are free to open PRs!